controller/query.go

```diff
@@ -0,0 +1,190 @@
+		}
+		res := &app.QuerySingle{
+			Data: ConvertQuery(appl, ctx.Request, q),
+		}
+		ctx.ResponseData.Header().Set("Location", rest.AbsoluteURL(ctx.Request, app.QueryHref(ctx.SpaceID, res.Data.ID)))
+		return ctx.Created(res)
+	})
+}
+
+// ConvertQuery converts from internal to external REST representation
+func ConvertQuery(appl application.Application, request *http.Request, q query.Query) *app.Query {
+	queryType := query.APIStringTypeQuery
+	spaceID := q.SpaceID.String()
+	relatedURL := rest.AbsoluteURL(request, app.QueryHref(spaceID, q.ID))
+	creatorID := q.Creator.String()
+	userType := APIStringTypeUser
+	relatedCreatorLink := rest.AbsoluteURL(request, fmt.Sprintf("%s/%s", usersEndpoint, creatorID))
+	// spaceRelatedURL := rest.AbsoluteURL(request, app.SpaceHref(spaceID))
+	appQuery := &app.Query{
+		Type: queryType,
+		ID:   &q.ID,
+		Attributes: &app.QueryAttributes{
+			Title:     q.Title,
+			Fields:    q.Fields,
+			CreatedAt: &q.CreatedAt,
+		},
+		Links: &app.GenericLinks{
+			Self:    &relatedURL,
+			Related: &relatedURL,
+		},
+		Relationships: &app.QueryRelations{
+			Creator: &app.RelationGeneric{
+				Data: &app.GenericData{
+					Type: &userType,
+					ID:   &creatorID,
+					Links: &app.GenericLinks{
+						Related: &relatedCreatorLink,
+					},
+				},
+			},
+		},
+	}
+	return appQuery
+}
+
+// ConvertQueries from internal to external REST representation
+func ConvertQueries(appl application.Application, request *http.Request, queries []query.Query) []*app.Query {
+	var ls = []*app.Query{}
+	for _, q := range queries {
+		ls = append(ls, ConvertQuery(appl, request, q))
+	}
+	return ls
+}
+
+// List runs the list action.
+func (c *QueryController) List(ctx *app.ListQueryContext) error {
+	currentUserIdentityID, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err.Error()))
+	}
+	return application.Transactional(c.db, func(appl application.Application) error {
+		_, err = appl.Spaces().Load(ctx, ctx.SpaceID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+		queries, err := appl.Queries().ListByCreator(ctx, ctx.SpaceID, *currentUserIdentityID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+		res := &app.QueryList{}
+		res.Data = ConvertQueries(appl, ctx.Request, queries)
+		return ctx.OK(res)
+	})
+}
+
+// Show runs the show action.
+func (c *QueryController) Show(ctx *app.ShowQueryContext) error {
+	return application.Transactional(c.db, func(appl application.Application) error {
+		_, err := appl.Spaces().Load(ctx, ctx.SpaceID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+		q, err := appl.Queries().Load(ctx, ctx.QueryID, ctx.SpaceID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+		res := &app.QuerySingle{
+			Data: ConvertQuery(appl, ctx.Request, *q),
+		}
+		return ctx.OK(res)
+	})
+}
+
+// Show runs the show action.
```

Comment:
Please update this comment for Delete action

Reply:
I think it is fixed meanwhile, right? @DhritiShikhar